Source/WebCore/bindings/js/JSDOMWrapper.h

```diff
@@ -77,6 +77,7 @@ class JSDOMWrapper : public JSDOMObject {
     using DOMWrapped = ImplementationClass;
 
     ImplementationClass& wrapped() const { return m_wrapped; }
+    Ref<ImplementationClass> protectedWrapped() const { return m_wrapped; }
```

Comment:
I guess eventually we may end up with three accessors per data member for certain data members: unprotected, protected, and weak. On the one hand, that feels a little ick. On the other hand, I think it's correct?

You could say that we should only have the protected accessor, but: If you already have a reference you don't want to waste compute on making another; and we probably can't ask the compiler to make that cost disappear; and if a class holds a weak pointer internally it can't guarantee a non-null return value.

You could say that we should only have the unprotected accessor, but: If you don't already have a reference, you don't want to have to write an extra line of code just to get one.

So I think this is how we want it to be.

Reply:
Note that none of this is new to this patch. Those are well established patterns since Ryosuke started smart pointer adoption. If you are not a fan of the current pattern, I suggest with start a Slack discussion with Ryosuke as he came up with this design and there is a lot of code impacted by it.

Reply:
I am saying that I am a fan of the pattern, but it took me some thinking to get there.

Reply:
A possible example of a weak pointer accessor is:

AccessibilityImageMapLink has a private data member

`    WeakPtr<HTMLAreaElement, WeakPtrImplWithEventTargetData> m_areaElement;
`
and a public accessor

`    HTMLAreaElement* areaElement() const { return m_areaElement.get(); }
`

Now that I think of it, I guess there's no case where the caller needs a WeakPtr accessor here. The caller can never chain the accessor into another call since they need to check for nullptr first. And if the caller wants to assign to their own WeakPtr, they can do so. 

A possible extreme example is if we have a hot loop that constructs WeakPtrs. In such a loop we might want a WeakPtr accessor because the WeakPtr copy constructor is slightly faster than the WeakPtr pointer constructor. I can't think of a case like that, though.